pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -303,6 +316,30 @@ def newpkgcb(pkgId, name, arch):
             result = await downloader.run()
             metadata_pb.increment()
 
+            repomd_path = result.path
+            repomd = cr.Repomd(repomd_path)
+
+            # Caution: we are not storing when the remote was last updated, so the order of this
+            # logic must remain in this order where we first check the version number as other
+            # changes than sync could have taken place such that the date or repo version will be
+            # different from last sync
+            if (
+                    optimize_sync
+                    and self.remote == self.repository.remote
+                    and (self.repository.last_sync_repo_version
```

Comment:
Try putting the operator on the line above and putting a ```\``` after it (the ```\``` might not be necessary with the parens though).

Reply:
Adding it to the flake8 config as an "ignore" would be acceptable too, if it doesn't work.  No need to stress over something like that :)  Linters are imperfect

Reply:
Not important but just to share my personal preference: I often assign some expressions and conditions to variables to make it more readable and to make some parts shorter.
E.g. I'd do something like this.
```
latest_version = self.repository.latest_version()
nothing_changed = (self.remote == self.repository.remote and
                   self.repository.last_sync_repo_version == latest_version.number and
                   self.remote.pulp_last_updated <= latest_version.pulp_created and
                   repomd.revision == self.repository.revision_number)
if optimize_sync and nothing_changed:
     ...
```

Since each condition is simple, I think it's fine if they are left as is. If they were complicated, one could go to the extreme like this (I'm not suggesting to do it for this PR, just an example):
```
latest_version = self.repository.latest_version()
is_same_remote = self.remote == self.repository.remote
is_same_latest_version = self.repository.last_sync_repo_version == latest_version.number
is_same_remote_config = self.remote.pulp_last_updated <= latest_version.pulp_created
is_same_revision = repomd.revision == self.repository.revision_number
nothing_changed = (is_same_remote and 
                   is_same_latest_version and  
                   is_same_remote_config and 
                   is_same_revision)
if optimize_sync and nothing_changed:
     ...
```